src/droplet.F

```diff
@@ -1932,15 +1941,83 @@ subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind,dpten)
       end if
 #endif
 
+
+
+      !!!!!!! HISTOGRAMS  !!!!!!!!
+
+      ! One last type of droplet stat is the histogram
+
+      bins_radius = 0.0
+      hist_radius = 0.0
+      hist_radius10 = 0.0
+
+      !First set up the bin centers
+      rmin = 1.0e-8
+      rmax = 1.0e-3
+      rmin10 = log10(rmin)
+      rmax10 = log10(rmax)
+
+      dr = (rmax10-rmin10)/(nbins-1)
+
+      bins_radius(1) = rmin10-dr
+      do i=1,nbins-1
+         bins_radius(i+1) = bins_radius(i) + dr
+      end do 
+
+      !Now that the bin centers are defined, loop through particles
+      do np=1,nparcelsLocalActive
+
+         rad10_tmp = log10(pdata(np,prrp))
+        
+         !Figure out which bin it belongs to
+         if (rad10_tmp .gt. rmax10 + 0.5*dr) then
+            ibin = nbins
+         elseif (rad10_tmp .lt. rmin10 - 0.5*dr) then
+            ibin = 1
+         else
+            ibin = (floor((rad10_tmp-(rmin10-0.5*dr))/dr)+1)+1
+         end if
+
+         if (ibin .gt. nbins .or. ibin .lt. 1) then
+         write (*,*) 'DHR BINS',rad10_tmp,rmin10,rmax10,dr,ibin
+         end if
+ 
+         !Increment that bin by the multiplicity
+         !(since that is how many droplets are said to have this size)
+         !hist_radius(ibin) = hist_radius(ibin) + pdata(np,prmult)
+         hist_radius(ibin) = hist_radius(ibin) + 1.0
+
+         if (pdata(np,prz) .lt. 10.0) then
+           hist_radius10(ibin) = hist_radius10(ibin) + 1.0
+         end if
+     
+      end do
+
+#ifdef MPI
+       !Sum up the histograms across MPI processes
+       if (myid == 0) then
+          call mpi_reduce(MPI_IN_PLACE,hist_radius,nbins,MPI_REAL, &
+                         MPI_SUM,0,MPI_COMM_WORLD,ierr)
+          call mpi_reduce(MPI_IN_PLACE,hist_radius10,nbins,MPI_REAL, &
+                         MPI_SUM,0,MPI_COMM_WORLD,ierr)
+       else
+          call mpi_reduce(hist_radius,hist_radius,nbins,MPI_REAL, &
+                         MPI_SUM,0,MPI_COMM_WORLD,ierr)
+          call mpi_reduce(hist_radius10,hist_radius10,nbins,MPI_REAL, &
+                         MPI_SUM,0,MPI_COMM_WORLD,ierr)
+       end if
+#endif
+
 #ifdef NETCDF
       ! JS: this subroutine is not updated with the new MPI
       !     interface for the droplets yet
 #ifdef MPI
       if ( myid == 0 ) then
 #endif
-      call writedropdiag_nc(nrec, &
+      call writedropdiag_nc(nrec,nbins, &
```

Comment:
so the `hist_radius10` and `hist_radius` are always written to the netCDF file? In this case, we could not check its values & correctness through the log files.

Reply:
That's right, however the average droplet radius is already being compared, which is the mean of this distribution. In other words, if the average droplet radius is checking out, then that must mean that the histogram is also okay. We could always add a write statement within this routine if needed, but the bigger thing is to figure out where to put the droplet_diag in the future